src/Destination/S3CmdDestination.php

```diff
@@ -112,7 +113,7 @@ public function all()
             throw new \RuntimeException($process->getErrorOutput());
         }
 
-        return $this->parseS3CmdListOutput($process->getOutput());
+        return new BackupCollection($this->parseS3CmdListOutput($process->getOutput()));
```

Comment:
No unfortunately. I only used flysystem to move files between local dirs, ssh, dropboxes and google drives so far. I imagine the library is slower that some potential alternative since there's quite a lot of abstraction in flysystem, but it's still kind of a standard for this kind of things our days.

But you're maybe right: if the performance in flysystem is a large concern (I personally don't the degree), it might be worth having other options in future (maybe as plugins?)